pkg/kubelet/kubelet_pods.go

```diff
@@ -1173,6 +1173,16 @@ func (kl *Kubelet) HandlePodCleanups(ctx context.Context) error {
 	metrics.RestartedPodTotal.WithLabelValues("true").Add(float64(restartCountStatic))
 	metrics.RestartedPodTotal.WithLabelValues("").Add(float64(restartCount))
 
+	// Complete termination of deleted pods that are not runtime pods (don't have
+	// running containers), are terminal, and are not known to pod workers.
+	// An example is pods rejected during kubelet admission that have never
+	// started before (i.e. does not have an orphaned pod).
+	// Triggering TerminatePod allows to proceed with force-deletion of such pods.
+	for _, pod := range kl.filterTerminalPodsToDelete(allPods, runningRuntimePods, workingPods) {
+		klog.V(3).InfoS("Completing termination of terminal pods scheduled for deletion", "pod", klog.KObj(pod), "podUID", pod.UID)
+		kl.statusManager.TerminatePod(pod)
```

Comment:
Is this racy?  Since pod worker is async, a pod may have been added to the pod worker, but not propogated to status manager.  Is the race prevented because we filter out pods known to the pod worker?  Is there any way for this resync process to race with pods completing in the pod worker?

Reply:
pods that aren't propagated to status manager would be eliminated by the GetPodStatus check. The main reason I have added it was to narrow down the scope just to the problematic pods. These are rejected pods, terminal, not added to pod workers, but added to status manager. I think skipping the check that the pod isn't it pod workers risks another scenario: that the pod's phase is changed to terminal by an external controller. In that case we still want to go through the syncTerminatingPod triggered by pod workers, so we don't want to interfere here in this scenario.

Reply:
This is outdated since now I call 
```
kl.podWorkers.UpdatePod(UpdatePodOptions{
			UpdateType: kubetypes.SyncPodKill,
			Pod:        pod,
		})
```
this completely eliminates the need to check status manager.